src/bindgen/parser.rs

```diff
@@ -156,6 +156,25 @@ impl<'a> Parser<'a> {
         assert!(self.lib.is_some());
         self.parsed_crates.insert(pkg.name.clone());
 
+        // Before we do anything with this crate, we must first parse all of its dependencies.
+        // This is guaranteed to terminate because the crate-graph is acyclic (and even if it
+        // wasn't, we've already marked the crate as parsed in the line above).
+        if self.lib.is_some() {
+            for (dep_pkg, cfg) in self.lib.as_ref().unwrap().dependencies(&pkg) {
+                if self.should_parse_dependency(&dep_pkg.name) {
```

Comment:
Would be cleaner as `if !self.should_parse_dependency() { continue; }` imo, less rightward drift. But your call.